public/app/plugins/datasource/grafana-azure-monitor-datasource/components/deprecated/components/InsightsConfig.tsx

```diff
@@ -1,25 +1,38 @@
+import {
+  updateDatasourcePluginJsonDataOption,
+  updateDatasourcePluginResetOption,
+  updateDatasourcePluginSecureJsonDataOption,
+} from '@grafana/data';
+import { Alert, Button, InlineFormLabel, LegacyForms } from '@grafana/ui';
 import React, { PureComponent } from 'react';
-import { InlineFormLabel, Button, LegacyForms, Alert } from '@grafana/ui';
+
+import { AzureDataSourceJsonData, AzureDataSourceSecureJsonData } from '../../../types';
+import { Props } from '../../ConfigEditor';
+
 const { Input } = LegacyForms;
-import { AzureDataSourceSettings, AzureDataSourceJsonData, AzureDataSourceSecureJsonData } from '../types';
 
-export interface Props {
-  options: AzureDataSourceSettings;
-  onUpdateJsonDataOption: (
-    key: keyof AzureDataSourceJsonData
-  ) => (event: React.SyntheticEvent<HTMLInputElement | HTMLSelectElement>) => void;
-  onUpdateSecureJsonDataOption: (
-    key: keyof AzureDataSourceSecureJsonData
-  ) => (event: React.SyntheticEvent<HTMLInputElement | HTMLSelectElement>) => void;
-  onResetOptionKey: (key: keyof AzureDataSourceSecureJsonData) => void;
-}
 export class InsightsConfig extends PureComponent<Props> {
   onAppInsightsResetApiKey = () => {
```

Comment:
can this also be private or am I missing the spot where we use it somewhere else?

Reply:
right, it was just like that but it can be private